### PR TITLE
[12.0]  MIGR sale_order_lot_generator

### PR DESCRIPTION
**Pull request by bealdav**
- [x] depends https://github.com/OCA/sale-workflow/pull/1143

### CODE REVIEW

File: sale_order_lot_generator/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    @api.model
+    def create_prodlot(self, index_lot=1):
+        vals = self._prepare_vals_lot_number(index_lot)
+        return self.env["stock.production.lot"].create(vals)
+
+    @api.model
+    def create(self, values):
+        line = self.new(values)
+        # we create a lot befor crete a line because the super method
```
Comment (apollet) [inline sale_order_lot_generator/models/sale.py]: typo:  we create a lot befor**e** cre**a**te a line

File: sale_order_lot_generator/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        # must create a procurement and move
+        if (
+            line.order_id.state == "sale"
+            and line.product_id.auto_generate_prodlot
+            and not line.lot_id
+            and line.product_id.tracking != "none"
+        ):
+            # wehen a new line is added to confirmed sale order
```
Comment (apollet) [inline sale_order_lot_generator/models/sale.py]: typo: **when** a new

**Review by apollet — APPROVED** [review]
LGTM just typo mistakes

**Comment by sebastienbeau** [comment]
LGTM code review

**Review by sebastienbeau — APPROVED** [review]
LGTM code review

**Comment by bealdav** [comment]
Thank you @apollet just fixed

**Comment by bealdav** [comment]
Hi @rousseldenis could you have a look here, travis is OK and migration is trivial, see last commit

Thanks

**Review by HaraldPanten — APPROVED** [review]
Hi @bealdav 

I suggest you to improve the README. For example, you could explain that you have to activate "Auto generate lot" checkbox.

Even though, this time is only a suggestion, do not hate me 😝

**Comment by bealdav** [comment]
@HaraldPanten I don't hate oca contributors (love them), and you're right it improves explanation and it's easy to update. Thanks

**Review by mourad-ehm — APPROVED** [review]

**Comment by bealdav** [comment]
ready to merge @rousseldenis, Thanks

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**